### PR TITLE
pkg/ansible: Add Phase to CR status field for Ansible Operator

### PR DESCRIPTION
**Pull request by dymurray**

### CODE REVIEW

File: pkg/ansible/controller/reconcile.go [hunk 1]
```diff
@@ -70,14 +70,35 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
...
 		u.Object["spec"] = map[string]interface{}{}
 		r.Client.Update(context.TODO(), u)
 		return reconcile.Result{Requeue: true}, nil
 	}
+	status := u.Object["status"]
+	_, ok = status.(map[string]interface{})
+	if !ok {
+		logrus.Debugf("status was not found")
+		u.Object["status"] = map[string]interface{}{}
+		r.Client.Update(context.TODO(), u)
```
Comment (hasbro17) [inline pkg/ansible/controller/reconcile.go]: Check return error.

File: pkg/ansible/controller/reconcile.go [hunk 2]
```diff
@@ -70,14 +70,35 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
...
+
+	// If status is an empty map we can assume CR was just created
+	if len(u.Object["status"].(map[string]interface{})) == 0 {
+		logrus.Debugf("Setting phase status to %v", StatusPhaseCreating)
+		u.Object["status"] = ResourceStatus{
+			Status: Status{
+				Phase: StatusPhaseCreating,
+			},
+		}
+		r.Client.Update(context.TODO(), u)
```
Comment (hasbro17) [inline pkg/ansible/controller/reconcile.go]: Check return error.

File: pkg/ansible/controller/types.go [hunk 3]
```diff
@@ -19,14 +19,18 @@ import (
...
+	StatusPhaseRunning  = "Running"
+	StatusPhaseFailed   = "Failed"
 )
 
 type Status struct {
 	Ok               int                `json:"ok"`
 	Changed          int                `json:"changed"`
 	Skipped          int                `json:"skipped"`
 	Failures         int                `json:"failures"`
+	Phase            string             `json:"phase"`
```
Comment (shawn-hurley) [inline pkg/ansible/controller/types.go]: I think the Phase should be on the ResourceStatus object below.

File: pkg/ansible/controller/reconcile.go [hunk 4]
```diff
@@ -70,14 +70,41 @@ func (r *AnsibleOperatorReconciler) Reconcile(request reconcile.Request) (reconc
...
+	if len(u.Object["status"].(map[string]interface{})) == 0 {
+		logrus.Debugf("Setting phase status to %v", StatusPhaseCreating)
+		u.Object["status"] = ResourceStatus{
+			Status: Status{
+				Phase: StatusPhaseCreating,
+			},
+		}
+		err = r.Client.Update(context.TODO(), u)
+		if err != nil {
+			return reconcile.Result{}, nil
```
Comment (shawn-hurley) [inline pkg/ansible/controller/reconcile.go]: I think we want to return the err here IMO

File: pkg/ansible/controller/types.go [hunk 5]
```diff
@@ -96,13 +99,15 @@ func NewStatusFromMap(sm map[string]interface{}) Status {
...
 	Status         `json:",inline"`
+	Phase          string   `json:"phase"`
 	FailureMessage string   `json:"reason,omitempty"`
 	History        []Status `json:"history,omitempty"`
 }
 
 func UpdateResourceStatus(sm map[string]interface{}, je eventapi.StatusJobEvent) (bool, ResourceStatus) {
 	newStatus := NewStatusFromStatusJobEvent(je)
 	oldStatus := NewStatusFromMap(sm)
+	phase := ""
```
Comment (shawn-hurley) [inline pkg/ansible/controller/types.go]: You could set phase = running and only change it if Failures > 0 and not have the else statement. I would suggest that we may want to do it that way.

**Comment by dymurray** [comment]
@shawn-hurley Let me know if you would like anything else changed.

**Comment by shawn-hurley** [comment]
@hasbro17 look good to you?

**Pull request merged**